sdk/core/azure-core/src/datetime.cpp

```diff
@@ -729,6 +744,30 @@ DateTime DateTime::Parse(std::string const& dateTime, DateFormat format)
       throw std::invalid_argument("Unrecognized date format.");
     }
 
+    if (adjustAsLocal)
+    {
+      auto const timeTNow = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+
+#ifdef _MSC_VER
+#pragma warning(push)
+// warning C4996: 'localtime': This function or variable may be unsafe. Consider using localtime_s
+// instead.
+#pragma warning(disable : 4996)
+#endif
+      // LCOV_EXCL_START
+      auto const utcDiffSeconds = localTimeToUtcDiffSeconds != nullptr
+          ? *localTimeToUtcDiffSeconds
+          : static_cast<int>(std::difftime(
+              std::mktime(std::localtime(&timeTNow)), std::mktime(std::gmtime(&timeTNow))));
+      // LCOV_EXCL_STOP
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
+
+      localDiffHours = static_cast<int8_t>(utcDiffSeconds / (60 * 60));
+      localDiffMinutes = static_cast<int8_t>((utcDiffSeconds % (60 * 60)) / 60);
```

Comment:
I am wary of such logic, especially if it implies daylight saving time isn't honored (and hence a token that is actually expired, isn't refreshed).

Is it only 2 hours within a year where this calculation would be wrong?